std/traits.d

```diff
@@ -8590,3 +8591,119 @@ enum isCopyable(S) = is(typeof(
+ +   (i.e. has no mutable indirections) AND is passed by value)
+ +   $(LI The Return Type is `immutable` OR can be impicitly converted to `immutable`)
+ + )
+ + Params:
+ + F = The function to be tested
+ +
+ + Returns:
+ + `true` if the function is strongly `pure`, `false` if otherwise.
+ + ++/
+template isStronglyPure(F...)
+if (F.length == 1 && isCallable!F)
+{
+    bool isStronglyPure()
+    {
+        alias STC = ParameterStorageClass;
+
+        if (!(functionAttributes!F & FunctionAttribute.pure_))
+            return false;
+
+        alias params = Parameters!F;
+
+        static foreach (i; 0 .. arity!F)
+        {
+            static if (is(params[i] == immutable))
+            {}
+            else static if (isRefType!(params[i]) || hasAliasing!(params[i])
+                    || ParameterStorageClassTuple!F[i] == STC.ref_
+                    || ParameterStorageClassTuple!F[i] == STC.out_)
+                return false;
+            else static if (isFunction!(params[i]))
+                if (!isStronglyPure!(params[i]))
+                    return false;
+        }
+
+        if (!is(ReturnType!F : immutable(ReturnType!F)))
+            return false;
+
+        return true;
+    }
+
+    private bool isRefType(T)()
+    {
+        if (is(T == class) || isDelegate!T || isAssociativeArray!T || isPointer!T)
+            return true;
+
+        return false;
+    }
+}
+
+///
+@safe pure unittest
+{
+    int f(int x) pure
+    {
+        return x + 1;
+    }
+
+    void g(immutable ref int x) pure
+    {}
+
+    void h(out int x) pure
+    {}
+
+    struct Test
+    {
+        int[string] aa;
+    }
+
+    void i(Test t) pure
+    {}
+
+    static assert( isStronglyPure!f);
+    static assert( isStronglyPure!g);
+    static assert(!isStronglyPure!h);
+    static assert(!isStronglyPure!i); //has parameter Test with mutable reference member aa
+}
+
+/++
+ + Determines if a function is a strongly `pure` function. To be strongly `pure`, a function must satisfy the following:
+ + $(UL
+ +   $(LI Each parameter is `immutable`, OR can be converted automatically to `immutable`
+ +   (i.e. has no mutable indirections) AND is passed by value)
+ +   $(LI The Return Type is `immutable` OR can be impicitly converted to `immutable`)
+ + )
+ + This version is useful if a function has multiple overloads. The correct overload can be selected by specifying
+ + the argument types as the template arguments
+ +
+ + Params:
+ + funcName = The name of the function to be tested
+ + T = The types of the arguments accepted by the overload that should be tested
+ +
+ + Example:
+ -----------------
+ void f() pure
+ {}
+ void f(out int x) pure
+ {}
+ static assert ( isStronglyPure!"f");
+ static assert (!isStronglyPure!("f", int));
+ -----------------
+ + Returns:
+ + `true` if the function is strongly `pure`, `false` if otherwise.
+ + ++/
+bool isStronglyPure(string funcName, T...)()
```

Comment:
Is is really useful ? I don't think. People are supposed to use `getOverloads` by themselves. Many traits for functions in this module don't have this helper.

Reply:
Also please combine the overloads with `/// ditto`, s.t. they are visually grouped on the docs.

Reply:
The parameters differ, i think this is why OP has copied the whole stuff.

Reply:
Considering the fact that it does not work for local functions, I think its best to get rid of this overload.